packages/scene-composer/src/hooks/useKeyPress.ts

```diff
@@ -0,0 +1,92 @@
+/* istanbul ignore file */
+// Credit original Source: https://codesandbox.io/s/rob7b8?file=/src/hooks/useKeyPress.tsx:0-2069
+// Heavily modified for my needs
+import { useCallback, useEffect, useState } from 'react';
+import { useFrame } from '@react-three/fiber';
+
+import { Key } from './useKeyMap';
+
+function isKey(key: string | number | Key): key is Key {
+  return !!(key as Key).key;
+}
+
+interface KeyPressArgs {
+  enabled?: boolean;
+  onKeyPressed?: (delta?: number) => void; // called every frame while key is held down.
```

Comment:
Nit-pick: Probably don't need this level of commenting as it is standard javascript behaviors for these actions.

Reply:
removed